UnityProject/Assets/Scripts/MapSaver/MapLoader.cs

```diff
@@ -1,13 +1,205 @@
+			{
+				MapSaver.CodeClass.ThisCodeClass.Reset(); //TODO Some of the functionality could be handled here
+			}
+
+			foreach (var prefabData in CompactObjectMapData.PrefabData)
+			{
+				ProcessIndividualObject(CompactObjectMapData, prefabData, Matrix, Offset00, Offset);
+			}
+
+
+
+			if (LoadingMultiple == false)
+			{
+				MapSaver.CodeClass.ThisCodeClass.Reset(); //TODO Some of the functionality could be handled here
+			}
+		}
+
+
+		public static void ProcessIndividualObject(MapSaver.CompactObjectMapData CompactObjectMapData, MapSaver.PrefabData prefabData, MatrixInfo Matrix, Vector3Int Offset00, Vector3Int Offset, GameObject Object = null)
+		{
+
+			if (Object == null)
+			{
+				MapSaver.StringToVector(prefabData.LocalPRS, out Vector3 position);
+				position += Offset00;
+				position += Offset;
+				Object = Spawn
+					.ServerPrefab(CustomNetworkManager.Instance.ForeverIDLookupSpawnablePrefabs[prefabData.PrefabID],
+						position.ToWorld(Matrix)).GameObject;
+				MapSaver.StringToPRS(Object, prefabData.LocalPRS);
+				Object.transform.localPosition += Offset00;
+				Object.transform.localPosition += Offset;
+				Object.GetComponent<UniversalObjectPhysics>()?.AppearAtWorldPositionServer(Object.transform.localPosition.ToWorld(Matrix));
+			}
+
+
+
+			if (string.IsNullOrEmpty(prefabData.Name) == false)
+			{
+				Object.name = prefabData.Name;
+			}
+
+			if (string.IsNullOrEmpty(prefabData.Name) == false)
+			{
+				Object.name = prefabData.Name;
+			}
+
+			MapSaver.CodeClass.ThisCodeClass.Objects[prefabData.GitID] = Object;
+			ProcessClassData(prefabData,Object, prefabData.Object);
+
+			if (CustomNetworkManager.IsServer)
+			{
+				CompactObjectMapData.IDToNetIDClient[prefabData.GitID] = Object.GetComponent<NetworkIdentity>().netId;
+			}
+
+			MapSaver.CodeClass.ThisCodeClass.FinishLoading(prefabData.GitID);
+		}
+
+		public static void ProcessClassData(MapSaver.PrefabData prefabData, GameObject Object, MapSaver.IndividualObject IndividualObject)
+		{
+			if (IndividualObject == null)
+			{
+				return;
+			}
+
+			if (string.IsNullOrEmpty(IndividualObject.Name) == false)
+			{
+				Object.name = IndividualObject.Name;
+			}
+
+			foreach (var classData in IndividualObject.ClassDatas)
+			{
+				var ClassComponents = classData.ClassID.Split("@"); //TODO Multiple components?
+				var Component = Object.GetComponent(ClassComponents[0]);
+				if (Component == null)
+				{
+					try
+					{
+						Component = Object.AddComponent(AllowedReflection.GetTypeByName(ClassComponents[0]));
+					}
+					catch (Exception e)
+					{
+						Loggy.LogError(e.ToString());
+						continue;
+					}
+				}
+
+				SecureMapsSaver.LoadData(prefabData.GitID , Component, classData.Data, MapSaver.CodeClass.ThisCodeClass);
+			}
+
+			if (IndividualObject.Children != null)
+			{
+				foreach (var Child in IndividualObject.Children)
+				{
+					var Id = int.Parse(Child.ID.Split(",").Last());
+					while (Object.transform.childCount <= Id)
```

Comment:
add a safety mechanism to this

Reply:
What would be something logical though, that Can't be just emulated with 100000000000 objects instead?